src/api/firebase.js

```diff
@@ -169,39 +169,87 @@ export async function shareList(listPath, currentUserId, recipientEmail) {
  * @param {string} itemData.itemName The name of the item.
  * @param {number} itemData.daysUntilNextPurchase The number of days until the user thinks they'll need to buy the item again.
  */
+
 export async function addItem(listPath, { itemName, daysUntilNextPurchase }) {
 	const listCollectionRef = collection(db, listPath, 'items');
-
 	const newDocRef = doc(listCollectionRef);
 
 	return setDoc(newDocRef, {
 		dateCreated: new Date(),
-		// NOTE: This is null because the item has just been created.
-		// We'll use updateItem to put a Date here when the item is purchased!
 		dateLastPurchased: null,
 		dateNextPurchased: getFutureDate(daysUntilNextPurchase),
 		name: itemName,
 		totalPurchases: 0,
 	});
 }
 
+/**
+ * Updates the specified item in the shopping list with the new purchase information.
+ * @param {string} listPath - Path of the shopping list in Firestore.
+ * @param {string} itemId - ID of the item to be updated.
+ * @param {Timestamp} dateLastPurchased - Timestamp of the last purchase date.
+ * @param {number} nextPurchaseEstimate - Estimated number of days until the next purchase.
+ * @returns {Promise} A promise that resolves when the item is successfully updated.
+ */
 export async function updateItem(
 	listPath,
 	itemId,
+	todaysDate,
 	dateLastPurchased,
+	dateNextPurchased,
 	nextPurchaseEstimate,
 ) {
 	const listCollectionRef = collection(db, listPath, 'items');
 
 	const itemDocRef = doc(listCollectionRef, itemId);
 
 	return updateDoc(itemDocRef, {
-		dateLastPurchased,
+		previousNextPurchased: dateNextPurchased,
+		previousLastPurchased: dateLastPurchased,
+		dateLastPurchased: todaysDate,
 		dateNextPurchased: getFutureDate(nextPurchaseEstimate),
 		totalPurchases: increment(1),
 	});
 }
 
+/**
+ * Removes the last purchase information from the specified item in the shopping list.
+ * If the item has no previous purchases, the function resolves without making any changes.
+ * @param {string} listPath - Path of the shopping list in Firestore.
+ * @param {string} itemId - ID of the item to be unchecked.
+ * @returns {Promise} A promise that resolves when the item is successfully unchecked or if there are no changes needed.
+ */
+
+export async function uncheckItem(
+	listPath,
+	itemId,
+	previousLastPurchased,
+	previousNextPurchased,
+) {
+	// Create a reference to the item document in the specified shopping list
```

Comment:
add `try catch `block`
` try {
        const listCollectionRef = collection(db, listPath, 'items'); 
....
}
    } catch (error) {
        console.error("Error unchecking item:", error);
        throw error; // Throw the error to handle it at the caller level
    }
`